lib/fake_aws/s3/request.rb

```diff
@@ -56,24 +40,16 @@ def has_key?
 
     private
 
-      def server_name
-        @server_name ||= @env["SERVER_NAME"]
-      end
-
-      def path_info
-        @path_info ||= @env["PATH_INFO"]
-      end
-
       def server_name_components
-        @server_name_components ||= server_name.split(".")
+        @server_name_components ||= host.split(".")
       end
 
       def path_components
-        @path_components ||= path_info.split("/").drop(1)
+        @path_components ||= path.split("/").drop(1)
       end
 
       def request_style
-        @request_style ||= if server_name =~ %r{s3[-\w\d]*\.amazonaws\.com$}
+        @request_style ||= if host =~ %r{s3[-\w\d]*}
```

Comment:
This change ensures a fake url for a rack app can be used to correctly identify the request style given that locally you can't/don't want to fake s3-region.amazonaws.com.